lib/elastic_apm/grpc.rb

```diff
@@ -5,15 +5,35 @@ module ElasticAPM
   class GRPC
     # @api private
     class ClientInterceptor < ::GRPC::ClientInterceptor
-      TYPE = 'external.grpc'
+      TYPE = 'external'
+      SUBTYPE = 'grpc'
 
       # rubocop:disable Lint/UnusedMethodArgument
       def request_response(request:, call:, method:, metadata:)
         return yield unless (transaction = ElasticAPM.current_transaction)
         if (trace_context = transaction.trace_context)
           trace_context.apply_headers { |k, v| metadata[k.downcase] = v }
         end
-        ElasticAPM.with_span(method, TYPE) do
+
+        peer = call.instance_variable_get(:@wrapped)&.peer
+        context =
+          if peer
+            split_peer = URI.split(peer)
+            destination = ElasticAPM::Span::Context::Destination.new(
+              type: TYPE,
+              name: SUBTYPE,
+              resource: peer,
+              address: split_peer[0],
+              port: split_peer[6]
+            )
+            ElasticAPM::Span::Context.new(destination: destination)
+          end
```

Comment:
I think it would be nice to extract this into its own method. This one's getting a bit long and hard to read.